contracts/libraries/LockDataLibrary.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.20;
+
+import {IPoolManager} from "../interfaces/IPoolManager.sol";
+
+/// @dev This library manages a custom storage implementation for a queue
+///      that tracks current lockers. The "sentinel" storage slot for this data structure,
+///      always passed in as IPoolManager.LockData storage self, stores not just the current
+///      length of the queue but also the global count of non-zero deltas across all lockers.
+///      The values of the data structure start at OFFSET, and each value is a locker address.
+library LockDataLibrary {
+    uint256 private constant OFFSET = uint256(keccak256("LockData"));
+
+    /// @dev Pushes a locker onto the end of the queue, and updates the sentinel storage slot.
+    function push(IPoolManager.LockData storage self, address locker) internal {
+        // read current value from the sentinel storage slot
+        uint128 length = self.length;
+        unchecked {
+            uint256 indexToWrite = OFFSET + length; // not in assembly because OFFSET is in the library scope
+            /// @solidity memory-safe-assembly
+            assembly {
+                // in the next storage slot, write the locker
+                sstore(indexToWrite, locker)
+            }
+            // update the sentinel storage slot
+            self.length = length + 1;
+        }
+    }
+
+    /// @dev Pops a locker off the end of the queue. Note that no storage gets cleared.
```

Comment:
Why don't we clear storage?

Reply:
oh yeah i meant to write more of an explanation here - as it stands we're not clearing it to not use the caller's storage refund up - in the final impl this is going to be transient so it doesn't matter

Reply:
cool and tstore 🔜

Reply:
I guess one odd behavior of not clearing though is that a getLock() may return you a stale locker if you input an index that has not been used in the current txn

Reply:
Don't see any issues with that per say, and won't be an issue w transient storage but idk worth noting?